api/src/test/java/org/openmrs/util/databasechange/Database2_2_0UpgradeIT.java

```diff
@@ -0,0 +1,115 @@
+	
+	public static final String TEST_DATA_DIR = "/org/openmrs/util/databasechange/";
+	
+	public static final String UPGRADE_TEST_2_1_0_TO_2_20_DATASET = TEST_DATA_DIR
+		+ "database2_1To2_20UpgradeTest-dataSet.xml";
+	
+	public static final String STANDARD_TEST_2_2_0_DATASET = TEST_DATA_DIR + "standardTest-2.2.0-dataSet.xml";
+	
+	public final static String DATABASE_PATH = TEST_DATA_DIR + "openmrs-2.2.0.h2.db";
+	
+	private DatabaseUpgradeTestUtil upgradeTestUtil;
+	
+	private static File testAppDataDir;
+	
+	@BeforeClass
+	public static void beforeClass() throws IOException {
+		testAppDataDir = File.createTempFile("appdir-for-unit-tests", "");
+		testAppDataDir.delete();// so we can make turn it into a directory
+		testAppDataDir.mkdir();
+		
+		System.setProperty(OpenmrsConstants.APPLICATION_DATA_DIRECTORY_RUNTIME_PROPERTY, testAppDataDir.getAbsolutePath());
+		OpenmrsUtil.setApplicationDataDirectory(testAppDataDir.getAbsolutePath());
+	}
+	
+	@AfterClass
+	public static void afterClass() throws Exception {
+		FileUtils.deleteDirectory(testAppDataDir);
+		//Just to be safe, not to affect other units in the test suite
+		System.clearProperty(OpenmrsConstants.APPLICATION_DATA_DIRECTORY_RUNTIME_PROPERTY);
+	}
+	
+	@Before
+	public void before() throws IOException, SQLException {
+		upgradeTestUtil = new DatabaseUpgradeTestUtil(DATABASE_PATH);
+	}
+	
+	@After
+	public void after() throws SQLException {
+		upgradeTestUtil.close();
+	}
+	
+	/**
+	 * This method creates mock order entry upgrade file
+	 *
+	 * @see org.openmrs.util.UpgradeUtil#getConceptIdForUnits(String)
+	 */
+	public static void createOrderEntryUpgradeFileWithTestData(String propString) throws IOException {
+		Properties props = new Properties();
+		props.load(new StringReader(propString));
+		String appDataDir = OpenmrsUtil.getApplicationDataDirectory();
+		File propFile = new File(appDataDir, DatabaseUtil.ORDER_ENTRY_UPGRADE_SETTINGS_FILENAME);
+		props.store(new FileWriter(propFile), null);
+		propFile.deleteOnExit();
+	}
+	
+	@Test
+	public void shouldHaveUpdatedOrderTableInDatabase() throws Exception {
+		String FULFILLER_STATUS = "RECEIVED";
+		String FULFILLER_COMMENT = "Comment";
+		
+		upgradeTestUtil.executeDataset(STANDARD_TEST_2_2_0_DATASET);
+		upgradeTestUtil.executeDataset(UPGRADE_TEST_2_1_0_TO_2_20_DATASET);
+		createOrderEntryUpgradeFileWithTestData("mg=111\ntab(s)=112\n1/day\\ x\\ 7\\ days/week=113\n2/day\\ x\\ 7\\ days/week=114");
+		
+		upgradeTestUtil.upgrade();
+		Connection connection = upgradeTestUtil.getConnection();
+		final String insertOrderQuery = "UPDATE orders SET fulfiller_status='"+ FULFILLER_STATUS + "' , fulfiller_comment = '" +FULFILLER_COMMENT + "' where order_id = 6 ";
+		DatabaseUtil.executeSQL(connection, insertOrderQuery, false);
+		connection.commit();
+		List<Map<String, String>> orders = upgradeTestUtil.select("orders", "order_id = 6 or order_id = 7 and fulfiller_comment != ''",
```

Comment:
I think you should exclude order ids from the where clause, you should select the order where status and comment match the values you set them to,

Reply:
Done